core/src/api/api.rs

```diff
@@ -417,11 +417,20 @@ impl GraphState {
 		}
 		// import bundles
 		for ImportBundle { schema_id, pages, dsnp_keys, dsnp_user_id, key_pairs } in payloads {
-			let connection_type = self
-				.environment
-				.get_config()
-				.get_connection_type_from_schema_id(*schema_id)
-				.ok_or(DsnpGraphError::InvalidSchemaId(*schema_id))?;
+			let connection_type_option =
+				self.environment.get_config().get_connection_type_from_schema_id(*schema_id);
+			let connection_type = match connection_type_option {
+				Some(connection_type) => connection_type,
+				None => {
+					if pages.is_empty() && (!key_pairs.is_empty() || !dsnp_keys.is_none()) {
+						// This condition implies that only keys are being imported
+						ConnectionType::Follow(PrivacyType::Private)
```

Comment:
I'm not sure about this, it doesn't look that clean

Reply:
I am divided here too, however this connection_type is not used anywhere in the rest of the logic under these conditions

We can skip it away if pages are empty? Looking into some cleaner solution

Reply:
Tweaked it a bit to be cleaner, let me know your thoughts